common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -509,8 +552,147 @@ bool base_wlan_hal_dwpal::attach_ctrl_interface(int vap_id)
 
 bool base_wlan_hal_dwpal::process_nl_events()
 {
-    LOG(ERROR) << __func__ << "isn't implemented by derived and shouldn't be called";
-    return false;
+    if (!m_dwpal_nl_ctx) {
+        LOG(ERROR) << "Invalid Netlink socket used for nl events (m_dwpal_nl_ctx == nullptr)";
+        return false;
+    }
+
+    // check if there is nothing to proccess
+    if (m_fd_nl_events <= 0) {
+        LOG(ERROR) << __func__ << "nothing to proccess fd= " << m_fd_nl_events;
+        return false;
+    }
+
+    // Passing a lambda with capture is not supported for standard C function
+    // pointers. As a workaround, we create a static (but thread local) wrapper
+    // function that calls the capturing lambda function.
+    static __thread std::function<DWPAL_Ret(struct nl_msg * msg)> nl_handler_cb_wrapper;
+    nl_handler_cb_wrapper = [&](struct nl_msg *msg) -> DWPAL_Ret {
+        if (!process_dwpal_nl_event(msg)) {
+            LOG(ERROR) << "User's netlink handler function failed!";
+            return DWPAL_FAILURE;
+        }
+        return DWPAL_SUCCESS;
+    };
+    auto nl_handler_cb = [](struct nl_msg *msg) -> DWPAL_Ret { return nl_handler_cb_wrapper(msg); };
+
+    //parsing will be done in callback function
+    if (dwpal_driver_nl_msg_get(m_dwpal_nl_ctx, DWPAL_NL_UNSOLICITED_EVENT, NULL, nl_handler_cb) ==
+        DWPAL_FAILURE) {
+        LOG(ERROR) << " dwpal_driver_nl_msg_get failed,"
+                   << " ctx=" << m_dwpal_nl_ctx;
+        return false;
+    }
+
+    return true;
+}
+
+bool base_wlan_hal_dwpal::dwpal_nl_cmd_set(const std::string &ifname, unsigned int nl_cmd,
+                                           unsigned char *vendor_data, size_t vendor_data_size)
+{
+    if (vendor_data == nullptr) {
+        LOG(ERROR) << __func__ << "vendor_data is NULL ==> Abort!";
+        return false;
+    }
+
+    if (dwpal_driver_nl_cmd_send(m_dwpal_nl_ctx, DWPAL_NL_UNSOLICITED_EVENT, (char *)ifname.c_str(),
+                                 NL80211_CMD_VENDOR, DWPAL_NETDEV_ID,
+                                 (enum ltq_nl80211_vendor_subcmds)nl_cmd, vendor_data,
+                                 vendor_data_size) != DWPAL_SUCCESS) {
+        LOG(ERROR) << __func__ << "ERROR for cmd = " << nl_cmd;
+        return false;
+    }
+
+    return true;
+}
+
+size_t base_wlan_hal_dwpal::dwpal_nl_cmd_get(const std::string &ifname, unsigned int nl_cmd,
+                                             unsigned char *out_buffer,
+                                             const size_t max_buffer_size)
+{
+
```

Comment:
Why wasn't this removed? Please remove before merging.

Reply:
Its the icsl clang issue .
I cant work with the Ubuntu since dwpal not build there. 
so i work from icsl and the clang there is different . 
I need to push my changes from icsl go to VM , run clang and push again .
Cant work with fix-ups like that and also its too much fuss.
I'll fix it after you approve.